Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -64,55 +66,43 @@ void USpatialActorChannel::Init(UNetConnection* InConnection, int32 ChannelIndex
 {
 	Super::Init(InConnection, ChannelIndex, bOpenedLocally);
 
-	SpatialNetDriver = Cast<USpatialNetDriver>(Connection->Driver);
-	check(SpatialNetDriver);
-
-	WorkerView = SpatialNetDriver->GetSpatialOS()->GetView();
+	NetDriver = Cast<USpatialNetDriver>(Connection->Driver);
+	check(NetDriver);
+	Sender = NetDriver->Sender;
+	Receiver = NetDriver->Receiver;
 }
 
 void USpatialActorChannel::DeleteEntityIfAuthoritative()
 {
-	bool bHasAuthority = false;
-	USpatialInterop* Interop = SpatialNetDriver->GetSpatialInterop();
-
-	TSharedPtr<worker::View> PinnedView = WorkerView.Pin();
-	if (PinnedView.IsValid())
+	if (NetDriver->Connection == nullptr)
 	{
-		bHasAuthority = Interop->IsAuthoritativeDestructionAllowed()
-			&& PinnedView->GetAuthority<improbable::Position>(ActorEntityId.ToSpatialEntityId()) == worker::Authority::kAuthoritative;
+		return;
 	}
 
-	UE_LOG(LogSpatialGDKActorChannel, Log, TEXT("Delete Entity request on %d. Has authority: %d "), ActorEntityId.ToSpatialEntityId(), bHasAuthority);
+	bool bHasAuthority = NetDriver->View->GetAuthority(EntityId, Position::ComponentId) == WORKER_AUTHORITY_AUTHORITATIVE;
+
+	UE_LOG(LogTemp, Log, TEXT("Delete entity request on %lld. Has authority: %d"), EntityId, (int)bHasAuthority);
 
 	// If we have authority and aren't trying to delete a critical entity, delete it
 	if (bHasAuthority && !IsCriticalEntity())
 	{
-		Interop->DeleteEntity(ActorEntityId);
+		Sender->SendDeleteEntityRequest(EntityId);
+		Receiver->CleanupDeletedEntity(EntityId);
 	}
 }
 
 bool USpatialActorChannel::IsCriticalEntity()
 {
 	// Don't delete if the actor is the spawner
-	if (ActorEntityId.ToSpatialEntityId() == SpatialConstants::EntityIds::SPAWNER_ENTITY_ID)
+	if (EntityId == SpatialConstants::EntityIds::SPAWNER_ENTITY_ID)
 	{
 		return true;
 	}
 
-	// Don't delete if the actor is a Singleton
-	PathNameToEntityIdMap* SingletonNameToEntityId = SpatialNetDriver->GetSpatialInterop()->GetSingletonNameToEntityId();
-
-	if (SingletonNameToEntityId == nullptr)
-	{
-		return false;
-	}
-
-	for(const auto& Pair : *SingletonNameToEntityId)
+	// Don't delete if singleton entity
+	if (NetDriver->GlobalStateManager->IsSingletonEntity(EntityId))
```

Comment:
We could just check against the Channel's actor's spatial flags here.

Reply:
Unfortunately you don't have the actor at this point in time.